src/main/java/com/treasuredata/client/TDClient.java

```diff
@@ -627,14 +637,78 @@ public void deleteBulkImportSession(String sessionName)
     public String startSavedQuery(String name, Date scheduledTime)
     {
         TDScheduleRunResult result =
-            doPost(
-                    buildUrl("/v3/schedule/run", name, Long.toString(scheduledTime.getTime() / 1000)),
-                    ImmutableMap.<String, String>of(),
-                    TDScheduleRunResult.class);
+                doPost(buildUrl("/v3/schedule/run", name, Long.toString(scheduledTime.getTime() / 1000)),
+                        TDScheduleRunResult.class);
         return result.getJobs().get(0).getJobId();
     }
 
     @Override
+    public List<TDSavedQuery> listSavedQueries()
+    {
+        return doGet(buildUrl("/v3/schedule/list"), TDSavedQuery.TDSavedQueryList.class).getSchedules();
+    }
+
+    protected String toJson(Object any)
+    {
+        try {
+            return httpClient.getObjectMapper().writeValueAsString(any);
+        }
+        catch (JsonProcessingException e) {
+            logger.error("Failed to produce json", e);
+            throw new TDClientException(TDClientException.ErrorType.INVALID_INPUT, String.format("Failed to create JSON string from %s", any));
+        }
+    }
+
+    @Override
+    public TDSavedQuery saveQuery(TDSaveQueryRequest query)
+    {
+        String json = toJson(query);
+        TDSavedQuery result =
+                doPost(
+                        buildUrl("/v3/schedule/create", query.getName()),
+                        ImmutableMap.<String, String>of(),
+                        Optional.of(json),
+                        TDSavedQuery.class);
+        return result;
+    }
+
+    @Override
+    public TDSavedQuery updateSavedQuery(String name, TDSavedQueryBuilder request)
```

Comment:
Maybe we should create TDSavedQueryUpdateRequest class to make it consistent, but due to the same reason in the above, we need to merge the existing saved query data and its diff. 

No concurrent control will be available in this API design. We should note this limitation (or fix TD-API side)